include/ruby/backward/2/attributes.h

```diff
@@ -60,7 +60,23 @@
 #define DEPRECATED_BY(n,x) RBIMPL_ATTR_DEPRECATED(("by: " # n)) x
 
 #undef DEPRECATED_TYPE
-#define DEPRECATED_TYPE(mseg, decl) decl RBIMPL_ATTR_DEPRECATED(mseg)
+#if defined(__GNUC__)
+# define DEPRECATED_TYPE(mesg, decl)                    \
+    _Pragma("message \"DEPRECATED_TYPE is deprecated, " \
+            "use RBIMPL_ATTR_DEPRECATED instead\"");    \
+    decl RBIMPL_ATTR_DEPRECATED(mseg)
+#elif defined(_MSC_VER)
+# pragma deprecated(DEPRECATED_TYPE)
+# define DEPRECATED_TYPE(mesg, decl)                              \
+    __pragma(message(__FILE__"("STRINGIZE(__LINE__)"): warning: " \
+                     "DEPRECATED_TYPE is deprecated, "            \
+                     "use RBIMPL_ATTR_DEPRECATED instead"))       \
+    decl RBIMPL_ATTR_DEPRECATED(mseg)
+#else
+# define DEPRECATED_TYPE(mesg, decl)                    \
+    <-<-"DEPRECATED_TYPE is deprecated, "               \
+        "use RBIMPL_ATTR_DEPRECATED instead"->->
+#endif
```

Comment:
Is `RBIMPL_ATTR_DEPRECATED` a public interface?
If non-gcc & non-VC errs, it should be an error even with gcc & VC?

Reply:
`RBIMPL_ATTR_DEPRECATED` is not a public interface.  The message needs improvements.

For warning vs. error, this code is a copy&paste of commit 41910c0d0e989c368c6e08690df8171589e52f76.